pkg/slack/client.go

```diff
@@ -284,6 +335,111 @@ func (c *Client) SetWorkspaceRole(ctx context.Context, teamID, userID, roleID st
 	return nil
 }
 
+// ListIDPGroups returns all IDP groups from the SCIM API.
+func (c *Client) ListIDPGroups(ctx context.Context) ([]GroupResource, error) {
+	groupsUrl, err := url.JoinPath(baseScimUrl, "Groups")
+	if err != nil {
+		return nil, err
+	}
+
+	var allGroups []GroupResource
+	startIndex := 1
+	count := 100
+
+	urlParams := url.Values{}
+	var res SCIMResponse[GroupResource]
+
+	for {
+		urlParams.Add("startIndex", fmt.Sprint(startIndex))
+		urlParams.Add("count", fmt.Sprint(count))
+		err = c.doRequest(ctx, groupsUrl, &res, http.MethodGet, nil, urlParams)
+		if err != nil {
+			return nil, fmt.Errorf("error fetching IDP groups: %w", err)
+		}
+
+		allGroups = append(allGroups, res.Resources...)
+
+		startIndex += res.ItemsPerPage
+
+		if res.TotalResults < startIndex {
+			break
+		}
+	}
+
+	return allGroups, nil
+}
+
+// GetIDPGroup returns a single IDP group from the SCIM API.
+func (c *Client) GetIDPGroup(ctx context.Context, groupID string) (*GroupResource, error) {
+	groupUrl, err := url.JoinPath(baseScimUrl, "Groups", groupID)
+	if err != nil {
+		return nil, err
+	}
+
+	var res GroupResource
+
+	err = c.doRequest(ctx, groupUrl, &res, http.MethodGet, nil, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error fetching IDP group: %w", err)
+	}
+
+	return &res, nil
+}
+
+type PatchOp struct {
+	Schemas    []string      `json:"schemas"`
+	Operations []ScimOperate `json:"Operations"`
+}
+
+type ScimOperate struct {
+	Op    string   `json:"op"`
+	Path  string   `json:"path"`
+	Value []UserID `json:"value"`
+}
+
+type UserID struct {
+	Value string `json:"value"`
+}
+
+// GroupPatch patches a group by adding or removing a user.
+func (c *Client) GroupPatch(ctx context.Context, groupID string, user string, remove bool) error {
+	groupSCIMUrl, err := url.JoinPath(baseScimUrl, "Groups", groupID)
```

Comment:
Is this url case sensitive? If not, it'd be nice to keep it lowercase.

Reply:
unfortunately, it is, just tested it